node/core/backing/src/lib.rs

```diff
@@ -0,0 +1,1720 @@
+			}
+			ValidationResult::Invalid => {
+				Statement::Invalid(candidate_hash)
+			}
+		};
+
+		self.issued_validity.insert(candidate_hash);
+
+		if let Some(signed_statement) = self.sign_statement(statement) {
+			self.distribute_signed_statement(signed_statement).await?;
+		}
+
+		Ok(v.0)
+	}
+
+	/// Import the statement and kick off validation work if it is a part of our assignment.
+	async fn maybe_validate_and_import(
+		&mut self,
+		statement: SignedFullStatement,
+	) -> Result<(), Error> {
+		if let Some(summary) = self.import_statement(&statement).await? {
+			if let Statement::Seconded(_) = statement.payload() {
+				if summary.group_id == self.assignment {
+					self.kick_off_validation_work(summary).await?;
+				}
+			}
+		}
+
+		Ok(())
+	}
+
+	fn sign_statement(&self, statement: Statement) -> Option<SignedFullStatement> {
+		let local_index = self.table_context.local_index()?;
+
+		let signing_key = self.table_context.key.as_ref()?;
+
+		let signed_statement = SignedFullStatement::sign(
+			statement,
+			&self.table_context.signing_context,
+			local_index,
+			signing_key,
+		);
+
+		Some(signed_statement)
+	}
+
+	fn check_statement_signature(&self, statement: &SignedFullStatement) -> Result<(), Error> {
+		let idx = statement.validator_index() as usize;
+
+		if self.table_context.validators.len() > idx {
+			statement.check_signature(
+				&self.table_context.signing_context,
+				&self.table_context.validators[idx],
+			).map_err(|_| Error::InvalidSignature)?;
+		} else {
+			return Err(Error::InvalidSignature);
+		}
+
+		Ok(())
+	}
+
+	async fn send_to_provisioner(&mut self, msg: ProvisionerMessage) -> Result<(), Error> {
+		self.tx_from.send(FromJob::Provisioner(msg)).await?;
+
+		Ok(())
+	}
+
+	async fn request_pov_from_distribution(
+		&mut self,
+		descriptor: CandidateDescriptor,
+	) -> Result<PoVBlock, Error> {
+		let (tx, rx) = oneshot::channel();
+
+		self.tx_from.send(FromJob::PoVDistribution(
+			PoVDistributionMessage::FetchPoV(self.parent, descriptor, tx)
+		)).await?;
+
+		Ok(rx.await?)
+	}
+
+	async fn request_candidate_validation(
+		&mut self,
+		candidate: AbridgedCandidateReceipt,
+		pov: PoVBlock,
+	) -> Result<(ValidationResult, GlobalValidationSchedule, LocalValidationData), Error> {
+		let (tx, rx) = oneshot::channel();
+
+		self.tx_from.send(FromJob::CandidateValidation(
+				CandidateValidationMessage::Validate(
+					self.parent,
+					candidate,
+					self.head_data.clone(),
+					pov,
+					tx,
+				)
+			)
+		).await?;
+
+		Ok(rx.await??)
+	}
+
+	async fn store_chunk(
+		&mut self,
+		id: ValidatorIndex,
+		chunk: ErasureChunk,
+	) -> Result<(), Error> {
+		self.tx_from.send(FromJob::AvailabilityStore(
+				AvailabilityStoreMessage::StoreChunk(self.parent, id, chunk)
+			)
+		).await?;
+
+		Ok(())
+	}
+
+	async fn make_pov_available(
+		&mut self,
+		pov_block: PoVBlock,
+		global_validation: GlobalValidationSchedule,
+		local_validation: LocalValidationData,
+	) -> Result<(), Error> {
```

Comment:
I am realizing now that `AvailabilityStore` should have a `StorePoV` as well, which might encapsulate this logic. It doesn't seem to truly belong in this module.

Reply:
https://github.com/paritytech/polkadot/issues/1240#issuecomment-655779515 Description of follow-up work that should come with a refactor of this code